#! /bin/sh -e
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: [PATCH] ppc32: fixes FTBFS on power3/4.
## DP: This patch doesn't build pmac_cache.S on power3/power4, since it is
## DP: broken there and not needed.
## DP: Patch author: Sven Luther <luther@debian.org>
## DP: Upstream status: FTBFS, submitted to linuxppc-dev and benh.

. $(dirname $0)/DPATCH

@DPATCH@
--- kernel-source-2.6.11/arch/ppc/platforms/Makefile.orig	2005-03-27 11:38:25.000000000 +0200
+++ kernel-source-2.6.11/arch/ppc/platforms/Makefile	2005-03-27 11:39:23.000000000 +0200
@@ -9,9 +9,15 @@
 ifeq ($(CONFIG_APUS),y)
 obj-$(CONFIG_PCI)		+= apus_pci.o
 endif
+ifeq ($(CONFIG_6xx),y)
 obj-$(CONFIG_PPC_PMAC)		+= pmac_pic.o pmac_setup.o pmac_time.o \
 					pmac_feature.o pmac_pci.o pmac_sleep.o \
 					pmac_low_i2c.o pmac_cache.o
+else
+obj-$(CONFIG_PPC_PMAC)		+= pmac_pic.o pmac_setup.o pmac_time.o \
+					pmac_feature.o pmac_pci.o pmac_sleep.o \
+					pmac_low_i2c.o
+endif
 obj-$(CONFIG_PPC_CHRP)		+= chrp_setup.o chrp_time.o chrp_pci.o \
 					chrp_pegasos_eth.o
 obj-$(CONFIG_PPC_PREP)		+= prep_pci.o prep_setup.o
